plugins/callback/diy.py

```diff
@@ -0,0 +1,1421 @@
+            self._output(spec=self._diy_spec)
+
+        if self._parent_has_callback():
+            with self._suppress_stdout(enabled=self._using_diy(spec=self._diy_spec)):
+                super(CallbackModule, self).v2_playbook_on_notify(handler, host)
+
+    def v2_playbook_on_no_hosts_matched(self):
+        self._diy_spec = self._get_output_specification(
+            loader=self._diy_loader,
+            variables=self._diy_spec['vars']
+        )
+
+        if self._using_diy(spec=self._diy_spec):
+            self._output(spec=self._diy_spec)
+
+        if self._parent_has_callback():
+            with self._suppress_stdout(enabled=self._using_diy(spec=self._diy_spec)):
+                super(CallbackModule, self).v2_playbook_on_no_hosts_matched()
+
+    def v2_playbook_on_no_hosts_remaining(self):
+        self._diy_spec = self._get_output_specification(
+            loader=self._diy_loader,
+            variables=self._diy_spec['vars']
+        )
+
+        if self._using_diy(spec=self._diy_spec):
+            self._output(spec=self._diy_spec)
+
+        if self._parent_has_callback():
+            with self._suppress_stdout(enabled=self._using_diy(spec=self._diy_spec)):
+                super(CallbackModule, self).v2_playbook_on_no_hosts_remaining()
+
+    def v2_playbook_on_task_start(self, task, is_conditional):
+        self._diy_task = task
+
+        self._diy_spec = self._get_output_specification(
+            loader=self._diy_loader,
+            variables=self._get_vars(
+                playbook=self._diy_playbook,
+                play=self._diy_play,
+                task=self._diy_task
+            )
+        )
+
+        if self._using_diy(spec=self._diy_spec):
+            self._output(spec=self._diy_spec)
+
+        if self._parent_has_callback():
+            with self._suppress_stdout(enabled=self._using_diy(spec=self._diy_spec)):
+                super(CallbackModule, self).v2_playbook_on_task_start(task, is_conditional)
+
+    # not implemented as the call to this is not implemented yet
+    def v2_playbook_on_cleanup_task_start(self, task):
+        pass
+
+    def v2_playbook_on_handler_task_start(self, task):
+        self._diy_task = task
+
+        self._diy_spec = self._get_output_specification(
+            loader=self._diy_loader,
+            variables=self._get_vars(
+                playbook=self._diy_playbook,
+                play=self._diy_play,
+                task=self._diy_task
+            )
+        )
+
+        if self._using_diy(spec=self._diy_spec):
+            self._output(spec=self._diy_spec)
+
+        if self._parent_has_callback():
+            with self._suppress_stdout(enabled=self._using_diy(spec=self._diy_spec)):
+                super(CallbackModule, self).v2_playbook_on_handler_task_start(task)
+
+    def v2_playbook_on_vars_prompt(self, varname, private=True, prompt=None, encrypt=None, confirm=False, salt_size=None, salt=None, default=None, unsafe=None):
```

Comment:
```suggestion
    def v2_playbook_on_vars_prompt(self, varname, private=True, prompt=None, encrypt=None,
                                   confirm=False, salt_size=None, salt=None, default=None, unsafe=None):
```
Let's make all such long lines a bit shorter where possible?